File: b_service.cpp
#include <iostream>
#include <string>
#include "grpcpp/grpcpp.h"
#include "verification.pb.h"
#include <cctype>

using namespace std;

using grpc::Channel;
using grpc::ClientContext;
using grpc::Status;
using verification::PhonePinRequest;
using verification::StatusResponse;
using verification::Verification;

class VerificationClient{
public:
	VerificationClient(shared_ptr<Channel> channel):stub_(Verification::NewStub(channel)){}

	bool CheckValidTelefon(string phone){
		if (isDigit(phone))
			return TRUE;
		else
			return FALSE;
	}
private:
	unique_ptr<Verification::Stub> stub_;
};


int main(){
	string target_str = "localhost:8002";
	VerificationClient verification(grpc::CreateChannel(target_str, grpc::InsecureChannelCredentials()));
	string reply = verification.CheckValidTelefon()

	return 0;
}

File: b_service.rb
require 'grpc'
require_relative 'verification_services_pb'

class VerificationService < Verification::Service
	def check_valid_telefon(request, _unused_call)
		if request.phone =~ /^7\d{10}/
			return StatusResponse.new(status: true)
		end
		StatusResponse.new(status: false)
	end
end

s = GRPC::RpcServer.new
s.add_http2_port('0.0.0.0:8002', :this_port_is_insecure)
s.handle(VerificationService)
s.run_till_terminated_or_interrupted([1, 'int', 'SIGQUIT'])